src/main/java/net/dries007/tfc/common/blockentities/BloomeryBlockEntity.java

```diff
@@ -307,56 +308,56 @@ private void addItemsFromWorld(int capacity)
         // If we don't have a recipe, we'll find the first recipe which matches one of the inputs, and assign that.
         // Then, assuming we do have a recipe, we'll re-check the inputs for any that can be added, and add up to an equal amount of both.
         final BlockPos internalPos = getInternalBlockPos();
-        final List<ItemEntity> items = level.getEntitiesOfClass(ItemEntity.class, new AABB(internalPos, internalPos.offset(1, BloomeryBlock.getChimneyLevels(level, internalPos) + 1, 1)), EntitySelector.ENTITY_STILL_ALIVE);
+        final List<ItemEntity> itemEntities = level.getEntitiesOfClass(ItemEntity.class, new AABB(internalPos, internalPos.offset(1, BloomeryBlock.getChimneyLevels(level, internalPos) + 1, 1)), EntitySelector.ENTITY_STILL_ALIVE);
 
         if (cachedRecipe == null)
         {
-            for (ItemEntity entity : items)
+            assert inputStacks.isEmpty(); // If the cached recipe is null, we must have no inputs
+
+            final Collection<BloomeryRecipe> recipes = Helpers.getRecipes(level, TFCRecipeTypes.BLOOMERY).values();
+            final ItemStackInventory inventory = new ItemStackInventory();
+
+            loop:
```

Comment:
is this java syntax...?

Reply:
Yeah, labels: https://docs.oracle.com/javase/tutorial/java/nutsandbolts/branch.html